### PR TITLE
leftover cards

### PR DESCRIPTION
**Pull request by aurel-l**
## Purpose
Add remaining cards
 - https://www.ebi.ac.uk/panda/jira/browse/TRM-25599
 - https://www.ebi.ac.uk/panda/jira/browse/TRM-25607
 - https://www.ebi.ac.uk/panda/jira/browse/TRM-25603

## Approach
Add remaining cards. Unify logic across card renderers in order to make similarities more obvious for a possible future refactor.

## Testing
Current tests still working, new tests for new card renderers

## Checklist
- [ ] My PR is scoped properly, and “does one thing only”
- [x] I have reviewed my own code
- [x] I have checked that linting checks pass and type safety is respected
- [x] I have checked that tests pass and coverage has at least improved, and if not explained the reasons why

Note that I've not covered the map-to links, as https://www.ebi.ac.uk/panda/jira/browse/TRM-25668 would come up with an approach that we could use in the same way in the result cards

### CODE REVIEW

File: src/supporting-data/taxonomy/components/results/TaxonomyCard.tsx [hunk 1]
```diff
@@ -15,16 +16,20 @@ import TaxonomyColumnConfiguration, {
...
   handleEntrySelection: (rowId: string) => void;
 }> = ({ data, selected, handleEntrySelection }) => {
   const history = useHistory();
 
+  const id = getIdKey(data);
+
   const handleCardClick = useCallback(() => {
-    history.push(getEntryPath(Namespace.taxonomy, data.taxonId));
-  }, [history, data.taxonId]);
+    history.push(getEntryPath(Namespace.taxonomy, id));
```
Comment (dlrice) [inline src/supporting-data/taxonomy/components/results/TaxonomyCard.tsx:31]: Any specific reason we're not using the `BLOCK_CLICK_ON_CARD` logic here?

File: src/proteomes/components/results/ProteomesCard.tsx [hunk 2]
```diff
@@ -28,16 +29,28 @@ const buscoColumnRenderer = ProteomesColumnConfiguration.get(
...
   handleEntrySelection: (rowId: string) => void;
 }> = ({ data, selected, handleEntrySelection }): JSX.Element => {
   const history = useHistory();
 
-  const handleCardClick = useCallback(() => {
-    history.push(getEntryPath(Namespace.proteomes, data.id));
-  }, [history, data.id]);
+  const id = getIdKey(data);
+
+  const handleCardClick = useCallback(
```
[the remark this answers is not in the document]
Comment (dlrice) [reply src/proteomes/components/results/ProteomesCard.tsx:45]: Going one step further: could we create a generalised component to handle all of the cards in the various namespaces?

[the remark this answers is not in the document]
Comment (aurel-l) [reply src/proteomes/components/results/ProteomesCard.tsx:45]: About the block click logic, we could, but it should probably do that on the franklin side of things anyway. Already mentionned that with Xav.
And agreed, this PR puts into spotlight a lot of refactoring possibilities, don't want to handle them in this PR though, but will create independent Jiras for these

File: src/proteomes/components/results/ProteomesCard.tsx [hunk 3]
```diff
@@ -28,16 +29,28 @@ const buscoColumnRenderer = ProteomesColumnConfiguration.get(
   ProteomesColumn.busco
 );
 
+const BLOCK_CLICK_ON_CARD = new Set(['A', 'INPUT', 'BUTTON']);
```
[the remark this answers is not in the document]
Comment (aurel-l) [reply src/proteomes/components/results/ProteomesCard.tsx:32]: https://github.com/ebi-uniprot/uniprot-website/pull/330#discussion_r604808774 so it's copied for now, but will have to move to franklin I believe

**Pull request merged**